src/waypoint.jsx

```diff
@@ -34,11 +34,24 @@ function getCurrentPosition(bounds) {
     return POSITIONS.invisible;
   }
 
+  // top is within the screen
   if (bounds.viewportTop <= bounds.waypointTop &&
       bounds.waypointTop <= bounds.viewportBottom) {
     return POSITIONS.inside;
   }
 
+  // bottom is within the screen
+  if (bounds.viewportTop <= bounds.waypointBottom &&
+      bounds.waypointBottom <= bounds.viewportBottom) {
+    return POSITIONS.inside;
+  }
+
+  // top is above the screen and bottom is below the screen
```

Comment:
Hmm... I'm slightly concerned about the possibility of bug introduction, and the above conditionals do not seem likely to be expensive operations, so I'd lean towards leaving as-is personally.